app/models/user_session.rb

```diff
@@ -1,6 +1,13 @@
 class UserSession < ApplicationRecord
 
-  TTL = 2.weeks.freeze
+  TTL = case Rails.configuration.three_scale.user_sesion_ttl
```

Comment:
@hallelujah , left it as constant for least destructive way for the change and leave refactoring for the time we decide to make configurable on a finer grain. Let me know if you are ok with that.

Reply:
I really prefer not to bother with those if cases

Those are settings, we are not supposed to guard against all things a user could do. Keep it simple

Reply:
@hallelujah , what change exactly do you suggest? To remove the `else` block or remove all conditions? If it is the latter, it will make IMO more complicated usage as all callers will need to do these checks instead of having them here. Or otherwise we'll have to change this to a method or something. Please let me know.

Reply:
yes just remove the check, this is not needed really to my opinion

```suggestion
  TTL = Rails.configuration.three_scale.user_sesion_ttl
```

Reply:
The 3scale operator usually prefers implementation where we prepare for env vars to be set with empty strings.

Reply:
This is different. While we have to do that more often, better have it in one place for all those settings. And not have sparse code that takes care of configuration values in multiple files. In fact I am really thinking of getting rid of such constants defined dynamically and have them gathered in one single `AppSettings` instance

Fair enough, I think this is OK for now but I'd suggest to make another PR to that, below a proposed approach:


```ruby
# lib/app_settings.rb
class AppSettings
   include Singleton

  attr_reader :config

  def initialize
     # Or just include ActiveSupport::Configurable ... everything works
    @config = ActiveSupport::Configurable::Configuration.new
    super
  end

  def self.configure(key, &block)
    instance.configure(key, &block)
  end

  def configure(key, &processor)
    config[key] = processor.call
  end

  def get(key)
    config.fetch key
  end

  def self.get(key)
    instance.get(key)
  end
end
```

Then you can use it in an initializer

```ruby
# config/initializers/configuration.rb
require 'app_settings'

AppSettings.configure('user_session.ttl') do
 ENV['USER_SESSION_TTL'].presence? || 2.weeks
end
```

That way we could probably get rid of the logic in the `config/*.yml` files and have a single way to require those configuration:

```ruby
 def remember_me   
   remember_me_for AppSettings.get('user_session.ttl')
end
```

Reply:
I like this idea, though I'd think of a path to make `AppSettings` a proper replacement for `Rails.settings.three_scale` in the future, rather than yet another settings system.

Reply:
Yes exactly what I was thinking. Juggling with different files inside of the config/application.rb might not be the proper way to do that.

I propose would be:

1. YAML config files should be free from logic, it is hard to override them with https://github.com/3scale/3scale-amp-openshift-templates/blob/baed04acb897f5d3d25ab5fddb2fcc3778c22a0d/amp/amp.yml#L2176-L2177
2. cleanup this mess https://github.com/3scale/porta/blob/61a168335181a09d1671da684a49f23078fe933d/config/application.rb#L143-L210
3. Try to put code logic where it belongs to, meaning in a class or initializer
  * not int the config/application.rb
4. Ensure the values are well formatted